### DISCUSSION TITLE
Is it possible to hook into x-for iterations?

### DISCUSSION THREAD

**Discussion by jfajardo5** [alpinejs/alpine#1164] [1. Help]
I'm looking to hook into each x-for addition so I can run some javascript just before a new element is rendered.

For example, I'd like to set timeouts before each new x-for template gets rendered.

I'm using TALL stack, and my application logic is on the server.

**Comment by aidenybai** [comment]
I can't think of any good way to do this - here's a possible workaround you could consider though:

Have two arrays: `currArray` and `refArray`. Add whatever content you need into the refArray, and add a watcher to `refArray`. The callback should contain code that incrementally adds the `refArray` content to the `currArray`, based on your setTimeout and you can add additional javascript in there. The `x-for` should point towards the `currArray`
